internal/docstore/drivertest/drivertest.go

```diff
@@ -526,6 +499,116 @@ func testCodec(t *testing.T, ct CodecTester) {
 		check(nt, &NT{}, ct.DocstoreEncode, ct.NativeDecode)
 		check(nt, &NT{}, ct.NativeEncode, ct.DocstoreDecode)
 	}
+
+}
+
+// Test decoding into an interface{}, where the decoder doesn't know the type of the
+// result and must return some Go type that accurately represents the value.
+// This is implemented by the AsInterface method of driver.Decoder.
+// Since it's fine for different providers to return different types in this case,
+// each test case compares against a list of possible values.
+func testBlindDecode(t *testing.T, ct CodecTester) {
+	if ct == nil {
+		t.Skip("no CodecTester")
+	}
+	t.Run("DocstoreEncode", func(t *testing.T) { testBlindDecode1(t, ct.DocstoreEncode, ct.DocstoreDecode) })
+	t.Run("NativeEncode", func(t *testing.T) { testBlindDecode1(t, ct.NativeEncode, ct.DocstoreDecode) })
+}
+
+func testBlindDecode1(t *testing.T, encode func(interface{}) (interface{}, error), decode func(_, _ interface{}) error) {
+	// Encode and decode expect a document, so use this struct to hold the values.
+	type S struct{ X interface{} }
+
+	for _, test := range []struct {
+		in    interface{} // the value to be encoded
+		want  interface{} // one possibility
```

Comment:
nit: use a []interface{} to hold possible results?

Reply:
I tried it that way first, and the tests were ugly because I had to write `[]interface{}` for every one, even the single-valued ones.